go/common/crypto/signature/signers/ledger/ledger_signer.go

```diff
@@ -94,7 +99,7 @@ func (fac *Factory) Load(role signature.SignerRole) (signature.Signer, error) {
 	if !ok {
 		return nil, fmt.Errorf("role %d is not supported when using the Ledger backed signer", role)
 	}
-	device, err := ledgerCommon.ConnectToDevice(fac.address)
+	device, err := ledgerCommon.ConnectToDevice(fac.address, append(pathPrefix, fac.index))
```

Comment:
Why does this need to be changed, the path prefix in `ConnectToDevice` is only used to identify a specific device so it shouldn't matter which derivation path is used for that part? There was some mention of the need to refactor this (see #3053).

Reply:
@kostko 
The standard (consumer) app and the validator app are completely different and independent apps.
`consumer: 44'/474'/.....`
`validator: 474'/474'/....`
based on the path we communicate to the correct app.
We could hide this behind some flag but I think it is better to keep ledger-oasis-go generic and let oasis-core to have the flexibility to use any path in the future?

Reply:
Yeah I agree, but this specific path in `ConnectToDevice` is just for discovering the device, so it doesn't really matter which path is used here? Because the only reason why we use a path at all is that there is apparently no other reason to configure a specific ledger device to be used (e.g., if we used something like a device index, in case multiple devices are connected, this wouldn't be deterministic)?

Reply:
`Path` is actually needed to infer the `appMode` (Consumer/Validator). Without that info you can't connect to the ledger (at least directly without any 'guessing') because some info (like CLA) changes in relation of the `appMode`.
Another option would be having a way to list all available devices, print their app addresses and also print if that device has a validator app installed. Then pass that address with the flag `--signer.ledger.address `.

Reply:
`pathPrefix` is needed by `ledger-oasis-go` in order to know to which app (validator or consumer) should do the connection to.